packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -191,9 +191,13 @@ library LibVerifying {
             config.blockMaxGasLimit == 0 ||
             config.maxTransactionsPerBlock == 0 ||
             config.maxBytesPerTxList == 0 ||
+            // EIP-4844 blob size up to 128K
+            config.maxBytesPerTxList > 128000 ||
             config.minTxGasLimit == 0 ||
             config.slotSmoothingFactor == 0 ||
             config.anchorTxGasLimit == 0 ||
+            // EIP-4844 blob deleted after 30 days
+            config.txListCacheExpiry > 30 * 24 hours ||
```

Comment:
I'm pretty sure we cannot know how old the data is with actual data blobs. I also don't think we should care, we have big problems otherwise. :)

Reply:
Blob data will be available for 30 days before they are deleted. In the worse case, if a blob is posted to the beacon chain 1 month ago, new block referencing it may not have DA at all. 

With calldata this is not a problem.

Reply:
But I don't think EIP-4844 even exposes a timestamp for when the data was committed, so old commitments will be usable and that's fine.

>  new block referencing it may not have DA at all.

If that would be true we would not be able to use EIP-4844. Ethereum itself doesn't guarantee the data to be there anymore, of course we still depend on ALL data to still be available forever, otherwise the rollup would not be secure anymore!

Reply:
> But I don't think EIP-4844 even exposes a timestamp for when the data was committed, so old commitments will be usable. And that's fine!
I don't think EIP-4844 explicitly exposes a timestamp **after** a blob is submitted, but we know the *current timestamp* in the blob transaction (or the transaction that can attach a blob). In otherword, I'm assuming the blob transaction is not a pure blob, but is capable of running custom contract that verifies the blob.

Reply:
> we still depend on ALL data to still be available forever, otherwise the rollup would not be secure anymore!

https://eips.ethereum.org/EIPS/eip-4844
<img width="1069" alt="Screenshot 2023-03-19 at 11 49 48" src="https://user-images.githubusercontent.com/99078276/226152540-59370b61-7a49-4186-adee-aa945b741506.png">

We need to figure out, post EIP4844, how to dupliciate blob data and make sure they are available for ever.

Reply:
> I don't think EIP-4844 explicitly exposes a timestamp after a blob is submitted, but we know the current timestamp in the blob transaction (or the transaction that can attach a blob). In otherword, I'm assuming the blob transaction is not a pure blob, but is capable of running custom contract that verifies the blob.

Maybe so, I could be wrong about this.

> Rollups need data to be available once, long enough to ensure honest actors can construct the rollup state, but not forever.

Yeah exactly, once it's made available by Ethereum for a short time it's up to others to store the data forever, because all users need to be able to get all data from the start of the chain.

> We need to figure out, post EIP4844, how to dupliciate blob data and make sure they are available for ever.

I don't think we really need to do anything special for this. There will be projects like the graph/etherscan that will store all this data, and the data will also be available for everyone running an L2 node, so users should have plenty of options.